File: module-Common/src/main/java/com/kingz/module/common/bean/MediaParams.java
package com.kingz.module.common.bean;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * author：KingZ
 * date：2019/7/31
 * description：媒体资源参数Bean
 */
public class MediaParams implements Parcelable {
    public static final String PARAMS_KEY = "playParams";
    private String videoId;
    private String videoType;
    private String videoUrl;
    private String videoName;

    public MediaParams() {
    }

    protected MediaParams(Parcel in) {
        videoId = in.readString();
        videoType = in.readString();
        videoUrl = in.readString();
        videoName = in.readString();
    }

    /**
     * 反序列化
     */
    public static final Creator<MediaParams> CREATOR = new Creator<MediaParams>() {
        @Override
        public MediaParams createFromParcel(Parcel in) {
            return new MediaParams(in);
        }

        @Override
        public MediaParams[] newArray(int size) {
            return new MediaParams[size];
        }
    };

    @Override
    public int describeContents() {
        return 0;
    }

    /**
     * 序列化过程
     */
    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(videoId);
        dest.writeString(videoType);
        dest.writeString(videoUrl);
        dest.writeString(videoName);
    }

    public void setVideoId(String videoId) {
        this.videoId = videoId;
    }

    public void setVideoUrl(String videoUrl) {
        this.videoUrl = videoUrl;
    }

    public String getVideoId() {
        return videoId;
    }

    public String getVideoUrl() {
        return videoUrl;
    }

    public String getVideoName() {
        return videoName;
    }

    public void setVideoName(String videoName) {
        this.videoName = videoName;
    }

    public String getVideoType() {
        return videoType;
    }

    public void setVideoType(String videoType) {
        this.videoType = videoType;
    }

    public enum VideoType {
        VOD, LIVE
    }

    public enum QualityType {
        LOW("流畅 180P"), STD("标清 270P"), HD("高清 480P"), SD("超清 720P"), UHD("蓝光 1080P");

        String qualityName;

        QualityType(String qualityName) {
            this.qualityName = qualityName;
        }

        public String getQualityName() {
            return qualityName;
        }

        public static QualityType value(String quality) {
            if ("LOW".equalsIgnoreCase(quality)) {
                return LOW;
            } else if ("STD".equalsIgnoreCase(quality)) {
                return STD;
            } else if ("HD".equalsIgnoreCase(quality)) {
                return HD;
            } else if ("SD".equalsIgnoreCase(quality)) {
                return SD;
            } else if ("4K".equalsIgnoreCase(quality)) {
                return UHD;
            }
            return STD;
        }
    }

}

File: module-Common/src/main/java/com/kingz/module/common/bean/MediaParams.kt
package com.kingz.module.common.bean

import android.os.Parcel
import android.os.Parcelable

/**
 * author：KingZ
 * date：2019/7/31
 * description：媒体资源参数Bean
 */
class MediaParams : Parcelable {
    var videoId: String? = null
    var videoType: String? = null
    var videoUrl: String? = null
    var videoName: String? = null
    var videoBkg: String? = null

    constructor()
    protected constructor(`in`: Parcel) {
        videoId = `in`.readString()
        videoType = `in`.readString()
        videoUrl = `in`.readString()
        videoName = `in`.readString()
        videoBkg = `in`.readString()
    }

    override fun describeContents(): Int {
        return 0
    }

    /**
     * 序列化过程
     */
    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(videoId)
        dest.writeString(videoType)
        dest.writeString(videoUrl)
        dest.writeString(videoName)
        dest.writeString(videoBkg)
    }

    enum class VideoType {
        VOD, LIVE
    }

    enum class QualityType(var qualityName: String) {
        LOW("流畅 180P"), STD("标清 270P"), HD("高清 480P"), SD("超清 720P"), UHD("蓝光 1080P");

        companion object {
            fun value(quality: String?): QualityType {
                if ("LOW".equals(quality, ignoreCase = true)) {
                    return LOW
                } else if ("STD".equals(quality, ignoreCase = true)) {
                    return STD
                } else if ("HD".equals(quality, ignoreCase = true)) {
                    return HD
                } else if ("SD".equals(quality, ignoreCase = true)) {
                    return SD
                } else if ("4K".equals(quality, ignoreCase = true)) {
                    return UHD
                }
                return STD
            }
        }

    }

    companion object CREATOR : Parcelable.Creator<MediaParams> {
        const val PARAMS_KEY = "playParams"

        override fun createFromParcel(parcel: Parcel): MediaParams {
            return MediaParams(parcel)
        }

        override fun newArray(size: Int): Array<MediaParams?> {
            return arrayOfNulls(size)
        }
    }
}